zerver/views/__init__.py

```diff
@@ -573,6 +573,11 @@ def get_context_data(self, **kwargs):
         except KeyError:
             pass
 
+        try:
+            context['next'] = self.request.GET['next']
+        except KeyError:
+            context['next'] = '/'
+
```

Comment:
I don't like this change, if we can avoid it, since it makes the login URL ugly unnecessarily.  What we've done with Zulip in general is assume the next page is `/` if `next` is not specified, see for example the `zulip_login_required` code.